tasks/ContactorFaultResetTask.cpp

```diff
@@ -0,0 +1,98 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "ContactorFaultResetTask.hpp"
+
+using namespace motors_weg_cvw300;
+using namespace linux_gpios;
+
+ContactorFaultResetTask::ContactorFaultResetTask(std::string const& name)
+    : ContactorFaultResetTaskBase(name)
+{
+}
+
+ContactorFaultResetTask::~ContactorFaultResetTask()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See ContactorFaultResetTask.hpp for more detailed
+// documentation about them.
+
+bool ContactorFaultResetTask::configureHook()
+{
+    if (!ContactorFaultResetTaskBase::configureHook())
+        return false;
+
+    m_propulsion_disabled_duration = _propulsion_disabled_duration.get();
+    m_number_of_attempts_to_change_controller_failure =
+        _number_of_attempts_to_change_controller_failure.get();
+    return true;
+}
+bool ContactorFaultResetTask::startHook()
+{
+    if (!ContactorFaultResetTaskBase::startHook())
+        return false;
+    m_number_of_contactor_faults = 0;
+    return true;
+}
+void ContactorFaultResetTask::updateHook()
+{
+    ContactorFaultResetTaskBase::updateHook();
+
+    GPIOState current_propulsion_enable_state;
+    if (_propulsion_enable_in.read(current_propulsion_enable_state) == RTT::NoData) {
+        return;
```

Comment:
If this happens, you should explicitly turn off propulsion by sending false to the propulsion enable output.